pkg/gardener/shoot/extender/provider/provider_azure_test.go

```diff
@@ -124,10 +126,34 @@ func TestProviderExtenderForPatchWorkersUpdateAzure(t *testing.T) {
 			ExpectedShootWorkers: fixMultipleWorkers([]workerConfig{
 				{"main-worker", "azure.small", "gardenlinux", "1312.4.0", 1, 3, []string{"1", "2", "3"}},
 				{"next-worker", "azure.large", "gardenlinux", "1312.2.0", 1, 3, []string{"1", "2", "3"}}}),
-			ExistingInfraConfig:        fixAzureInfrastructureConfig("10.250.0.0/22", []string{"1", "2", "3"}),
+			ExistingInfraConfig:        fixAzureInfrastructureConfig(t, "10.250.0.0/22", []string{"1", "2", "3"}),
 			ExistingControlPlaneConfig: fixAzureControlPlaneConfig(),
 		},
-		"Add additional worker - extend existing additional worker from non HA setup to HA setup by adding more zones, infrastructureConfig already has three zones": {
+		"Add additional worker with new zone": {
```

Comment:
This is very good - exactly this use-case is expected when we support GPU-cases where just a few zones are offering particular GPU machines

Wouldn't it make sense to add this test-case also for AWS as we have similar situation there?